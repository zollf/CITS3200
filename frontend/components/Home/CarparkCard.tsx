import React from "react"
import styles from './styles.module.css';
import Map from '@/app/resources/static/images/map-pin.svg'
import NewWindow from '@/app/resources/static/images/open-in-new.svg'

const CarparkCard = () => {
    return (
        <div className={styles.card}>
            <div className={styles.topHalf}>
                Admin Carpark North
            </div>
            <div className={styles.bottomHalf}>
                <div className={styles.description}>
                    Car park located somewhere.
                </div>
                <div className={styles.directions}> 
                    <Map />

                    <a href="https://goo.gl/maps/RytgNDB4MW8McDBY6" target="_blank">
                        &nbsp; Get directions
                        <NewWindow />
                    </a>
                </div>
            </div>
        </div>
    );
};
  
export default CarparkCard;