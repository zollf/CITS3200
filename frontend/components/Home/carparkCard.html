<!DOCTYPE html>
<html lang="en-US">

<head>
    <title>practice</title>
    <meta charset="UTF-8">
    <link href="styles.css" rel="stylesheet" type="text/css">
    <link href='https://fonts.googleapis.com/css?family=Roboto' rel='stylesheet'>
    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
</head>

<body>
    <div class="card">
        <div class="topHalf">
            Admin Carpark North
        </div>
        <div class="bottomHalf">
            <div class="description">
                Car park located somewhere.
            </div>
            <div class="directions"> 
                <i class="material-icons" style="font-size:20px">room</i>

                <a href="https://goo.gl/maps/RytgNDB4MW8McDBY6" target="_blank">
                    &nbsp; Get directions
                    <i class="material-icons newWindowIcon">open_in_new</i>
                </a>
            </div>
        </div>
    </div>
</body>