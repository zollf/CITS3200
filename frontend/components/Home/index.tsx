import React from 'react';
import Tick from '@/app/resources/static/images/done.svg';
import styles from './styles.module.css';

const Home = () => {
  return (
    <div className={styles.home}>
      Hello from React! 2.0
      <Tick />
    </div>
  );
};

export default Home;
