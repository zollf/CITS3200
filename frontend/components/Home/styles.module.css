.home {
  color: red;
}

.card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  width: 300px;
  height: 221px;
  
  border: 3px solid #27348B;
  border-radius: 7px;
  background-color: #27348b;

  margin: auto;
}

.card:hover {
box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}

.topHalf {
  /* Top Half Card Container */
  box-sizing: border-box;
  width: 300px;
  height: 100px;
  text-align: center;
  display: table-cell;
  vertical-align: middle;

  /* Title Font */
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 28px;
  line-height: 33px;
  color: white;
}

.bottomHalf {
  box-sizing: border-box; 
  width: 300px;
  height: 121px;
  background-color: white;
  border-top: 3px solid #FDC300;
  border-radius: 0px 0px 7px 7px;
}

.description {
  /* Description Container */
  box-sizing: border-box;
  width: 300px;
  height: 79px;
  /* background-color:green; */

  /* Description Font */
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 19px;
  padding: 10px 20px;
}

.directions {
  /* Directions Container */
  box-sizing: border-box;
  width: 300px;
  height:39px;
  border-radius: 0px 0px 7px 7px;
  /* background: red; */
  
  /* Directions Font */
  font-family: Roboto;
  font-style: normal;
  font-weight: bold;
  font-size: 12px;
  /* line-height: 39px; */
  padding: 0px 20px;
  display: flex;
  align-items: center;
}

a:link {
  color: inherit;
  text-decoration: none;
  align-self: inherit;
}
a:visited {
  color: inherit;
}
a:hover {
  color: #1488ca;
}
a:active {
  color: #FDC300;
}

i.newWindowIcon {
  font-size: 10px;
  color: #323232;
}
i.newWindowIcon:hover {
  color: #1488ca;
} 
/* Ask about this */