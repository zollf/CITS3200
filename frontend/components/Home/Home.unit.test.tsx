import Home from './';
import React from 'react';
import { render } from '@testing-library/react';

it('matches its snapshot', () => {
  expect(render(<Home />).asFragment()).toMatchSnapshot();
});
