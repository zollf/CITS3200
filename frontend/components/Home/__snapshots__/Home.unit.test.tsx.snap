// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches its snapshot 1`] = `
<DocumentFragment>
  <div>
    Hello from React!
  </div>
</DocumentFragment>
`;
